Add StorCli text collector example script

mattbostock opened the pull request:
Collect metrics from the StorCLI utility on the health of MegaRAID
hardware RAID controllers and write them to stdout so that they can be
used by the textfile collector.

We parse the JSON output that StorCLI provides.

Script must be run as root or with appropriate capabilities for storcli
to access the RAID card.

Designed to run under Python 2.7, using the system Python provided with
many Linux distributions.

The metrics look like this:

```
mbostock@host:~$ sudo ./storcli.py
megaraid_status_code 0
megaraid_controllers_count 1
megaraid_emergency_hot_spare{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_scheduled_patrol_read{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_virtual_drives{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_drive_groups{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_virtual_drives_optimal{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_degraded{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 0
megaraid_battery_backup_healthy{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_ports{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 8
megaraid_failed{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 0
megaraid_drive_groups_optimal{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_healthy{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 1
megaraid_physical_drives{controller="0", model="AVAGOMegaRAIDSASPCIExpressROMB"} 24
mbostock@host:~$
```

I don't code regularly in Python so any suggestions for improvements welcome.

brian-brazil commented on text_collector_examples/storcli.py:
```diff
@@ -0,0 +1,73 @@
+
+
+def get_store_cli_json():
+    storcli_cmd = ['/opt/MegaRAID/storcli/storcli64', 'show', 'all', 'J']
```
This path needs to be configurable

mattbostock replied:
I left it hardcoded as I figured someone using this would likely copy and adapt it (as an example), but agree it's nicer to make it configurable. Would you prefer a commandline flag?

brian-brazil commented on text_collector_examples/storcli.py:
```diff
@@ -0,0 +1,73 @@
+                }
+
+        for name, value in controller_metrics.iteritems():
+            print("{}{}{} {}".format(METRIC_PREFIX, name, labels, value))
```
This is only writign to stdout, it should take care of creating the file and atomically changing it

SuperQ replied:
I agree, I think we should have a standard interface for all of these scripts.  I also would prefer stdout and not have to implement atomic writing in every script, and have a simple atomic write wrapper that can be used to put the metrics files in the correct place.

brian-brazil commented:
Also model doesn't belong as a label, the controller number is sufficient to identify the controller. I'd suggest using the machine role approach for it.

mattbostock commented:
I hesitated to rely on the controller number as I didn't know how stable it is. Happy to remove the model label.

brian-brazil commented:
If it's not stable, the model likely doesn't help you as the chances are you have identical models in a given machine.

SuperQ commented:
I would recommend a `megaraid_controller_info` metric with the `controller` and `model` labels.  This allows for annotation without having the `model` label on every metric.

SuperQ commented:
Ping, any progress on finishing this?

mattbostock commented:
Sorry for the delay, will get to this soon.

mattbostock commented:
@discordianfish @brian-brazil @SuperQ: I've amended to address the comments in this PR:

- added a `--storcli_path` option to set the path to storcli
- moved the model label to a megaraid_controller_info metric

I've also added `--help` and `--version` flags, and changed the logic so that the script fails more gracefully on machines where no MegaRAID cards are installed.

Any suggestions for changes welcome.
